tests/test_resource_tracker.py

```diff
@@ -0,0 +1,220 @@
+        def unregister(name, rtype):
+            # resource_tracker.unregister is actually a bound method of the
+            # ResourceTracker. We need a custom wrapper to avoid object
+            # serialization.
+            from loky.backend import resource_tracker
+            resource_tracker.unregister(folder_name, rtype)
+
+        e = ProcessPoolExecutor(1)
+        e.submit(unregister, folder_name, "folder").result()
+        e.shutdown()
+        '''
+        try:
+            p = subprocess.Popen(
+                [sys.executable, '-E', '-c', cmd.format(folder_name)],
+                stderr=subprocess.PIPE)
+            p.wait()
+
+            err = p.stderr.read().decode('utf-8')
+            p.stderr.close()
+
+            assert re.search("unregister %s" % folder_name, err) is not None
+            assert re.search("KeyError: '%s'" % folder_name, err) is None
+
+        finally:
+            executor.shutdown()
+
+    # The following four tests are inspired from cpython _test_multiprocessing
+    @pytest.mark.parametrize("rtype", ["folder", "semlock"])
+    def test_resource_tracker(self, rtype):
+        #
+        # Check that killing process does not leak named resources
+        #
+        if (sys.platform == "win32") and rtype == "semlock":
+            # no semlock on windows
+            return
+
+        import subprocess
+        cmd = '''if 1:
+            import time, os, tempfile, sys
+
+            from loky.backend.semlock import SemLock
+            from loky.backend import resource_tracker, reduction
+
+            def create_resource(rtype):
+                if rtype == "folder":
+                    return tempfile.mkdtemp(dir=os.getcwd())
+
+                elif rtype == "semlock":
+                    name = "/loky-%i-%s" % (os.getpid(), next(SemLock._rand))
+                    lock = SemLock(1, 1, 1, name)
+                    return name
+                else:
+                    raise ValueError(
+                        "Resource type %s not understood" % rtype)
+
+            for _ in range(2):
+                rname = create_resource("{rtype}")
+                resource_tracker.register(rname, "{rtype}")
+                # give the resource_tracker time to register the new resource
+                time.sleep(0.5)
+                sys.stdout.write(rname + "\\n")
+                sys.stdout.flush()
+            time.sleep(10)
+        '''
+        cmd = cmd.format(rtype=rtype, parent_pid=os.getpid())
+        p = subprocess.Popen([sys.executable, '-E', '-c', cmd],
+                             stderr=subprocess.PIPE,
+                             stdout=subprocess.PIPE)
+        name1 = p.stdout.readline().rstrip().decode('ascii')
+        name2 = p.stdout.readline().rstrip().decode('ascii')
+
+        # subprocess holding a reference to lock1 is still alive, so this call
+        # should succeed
+        _resource_unlink(name1, rtype)
+        p.terminate()
+        p.wait()
+        time.sleep(2.0)
```

Comment:
Add a comment to say that this timeout is needed to let the `ressource_tracker` the time to end.
To make this test more stable, couldn't you retrieve the semaphore process `pid` and wait is?

Reply:
> couldn't you retrieve the semaphore process pid and wait is?

No, we cannot wait on a grand-child process.